compiler/type-hint.h

```diff
@@ -454,12 +457,18 @@ class TypeHintShape : public TypeHint {
     , items(items)
     , is_vararg(is_vararg) {}
 
+  static std::map<std::int64_t, std::string> shape_keys_storage_;
+  static std::mutex shape_keys_storage_lock_;
+
 public:
   std::vector<std::pair<std::string, const TypeHint *>> items;
-  bool is_vararg;
+  bool is_vararg{false};
 
   static const TypeHint *create(std::vector<std::pair<std::string, const TypeHint *>> &&items, bool is_vararg);
 
+  static void set_shape_key(std::int64_t key_hash, std::string_view key) noexcept;
```

Comment:
1) There is some better naming for `set_shape_key`, since it doesn't set any shape key, it adds a known key to storage. Maybe, something like `register_known_key`? And not `get_shape_keys_storage`, but `get_all_registered_keys`?
2) We store std::string in a map, let's take std::string here also, not std::string_view

Reply:
1) Done
2) disagree, string_view intentionally used here to make code simple and clean